core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/GlrRenderTarget.java

```diff
@@ -275,89 +271,68 @@ private  double getFar(AbstractCamera sgCamera) {
     }
   }
 
-  private Vector4 transformFromViewportToProjection_AffectReturnValuePassedIn(Vector4 rv, AbstractCamera sgCamera, Rectangle actualViewport) {
-    rv.multiply(1.0 / rv.w);
+  private Vector4 transformFromViewportToProjection(Vector4 viewportCoord, AbstractCamera sgCamera, Rectangle actualViewport) {
+    Vector4 scaled = viewportCoord.dividedBy(viewportCoord.w());
 
-    rv.x = (rv.x - actualViewport.x) / actualViewport.width;
-    rv.y = (rv.y - actualViewport.y) / actualViewport.height;
+    double x = (scaled.x() - actualViewport.x) / actualViewport.width;
+    double y = (scaled.y() - actualViewport.y) / actualViewport.height;
 
     double zNear = getNear(sgCamera);
     double zFar = getFar(sgCamera);
-    rv.z = (rv.z - zNear) / (zFar - zNear);
-
-    rv.x = (rv.x * 2.0) - 1.0;
-    rv.y = (rv.y * 2.0) - 1.0;
-    rv.z = (rv.z * 2.0) - 1.0;
+    double z = (scaled.z() - zNear) / (zFar - zNear);
 
-    return rv;
+    return new Vector4((x * 2.0) - 1.0, (y * 2.0) - 1.0, (z * 2.0) - 1.0, 1.0);
   }
 
-  private Vector4 transformFromProjectionToViewport_AffectReturnValuePassedIn(Vector4 rv, AbstractCamera sgCamera, Rectangle actualViewport) {
-    rv.multiply(1.0 / rv.w);
+  private Vector4 transformFromProjectionToViewport(Vector4 projectionCoord, AbstractCamera sgCamera, Rectangle actualViewport) {
+    Vector4 scaled = projectionCoord.dividedBy(projectionCoord.w());
 
-    rv.x = (rv.x * 0.5) + 0.5;
-    rv.y = (rv.y * 0.5) + 0.5;
-    rv.z = (rv.z * 0.5) + 0.5;
+    double x = (scaled.x() * 0.5) + 0.5;
+    double y = (scaled.y() * 0.5) + 0.5;
+    double z = (scaled.z() * 0.5) + 0.5;
 
-    rv.x = (rv.x * actualViewport.width) + actualViewport.x;
-    rv.y = (rv.y * actualViewport.height) + actualViewport.y;
+    x = (x * actualViewport.width) + actualViewport.x;
+    y = (y * actualViewport.height) + actualViewport.y;
     double zNear = getNear(sgCamera);
     double zFar = getFar(sgCamera);
-    rv.z = (rv.z * (zFar - zNear)) + zNear;
-
-    return rv;
-  }
+    z = (z * (zFar - zNear)) + zNear;
 
-  private  Vector4 transformFromProjectionToCamera_AffectReturnValuePassedIn(Vector4 rv, AbstractCamera sgCamera) {
-    rv.multiply(1.0 / rv.w);
-    Matrix4x4 actualMatrix = getActualProjectionMatrix(sgCamera);
-    actualMatrix.invert();
-    actualMatrix.transform(rv);
-    return rv;
+    return new Vector4(x, y, z, 1.0);
   }
 
-  private  Vector4 transformFromCameraToProjection_AffectReturnValuePassedIn(Vector4 rv, AbstractCamera sgCamera) {
-    rv.multiply(1.0 / rv.w);
-    Matrix4x4 actualMatrix = getActualProjectionMatrix(sgCamera);
-    actualMatrix.transform(rv);
-    return rv;
+  private Vector4 transformFromProjectionToCamera(Vector4 projectionCoord, AbstractCamera sgCamera) {
+    return getActualProjectionMatrix(sgCamera).invert().transform(projectionCoord);
```

Comment:
this used to be scaled, was that unnecessary?

Reply:
By scaled do you mean the division by w?
If so, it was always 1 so I dropped it.